packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -584,12 +585,19 @@ const DataGrid = injectIntl((props) => {
       // Not triggered when swapping focus between children
       if (handleFocus.current) {
         setFocusedRowCol(focusedRow, focusedCol, true);
+        setGridHasFocus(true);
       }
     }
 
     handleFocus.current = true;
   };
 
+  const onBlur = (event) => {
+    if (!event.currentTarget.contains(event.relatedTarget)) {
+      setGridHasFocus(false);
+    }
+  };
```

Comment:
We should not need to add new event handler logic to determine if the grid has focus.

Reply:
But the column header cell would lose focus itself.  We should not need a state update because the CSS is based on focus, right?